### ISSUE TITLE
Connection timeouts don't close underlying socket

### ISSUE THREAD

**Issue by dhensby** [tediousjs/tedious#1157]
When creating a tedious connection, if the connection timeout elapses the underlying socket is still left open, rather than being closed.

This can cause the node process to hang if the socket hasn't closed itself by the time the node process is due to exit.

Please see https://github.com/tediousjs/node-mssql/issues/1106 for discussions relating to this.

**Comment by IanChokS** [comment]
Thanks for pointing this issue out, we'll take a look at it as soon as possible!

**Comment by IanChokS** [comment]
This looks like an issue related to #1100. Unfortunately, that hasn't had any follow ups yet either.

**Comment by dhensby** [comment]
Yep - that looks like a very well articulated look into what's going on and if that were resolved I think this issue would be too
